### PR TITLE
access matrices as artifacts

### PR DESCRIPTION
**Pull request by dpo**
* matrices in mat format are no longer supported;
* matrix_path and group_path are now obsolete;
* add script to generate Artifacts.toml;
* update unit tests;
* update README and documentation.

### CODE REVIEW

File: src/SuiteSparseMatrixCollection.jl [hunk 1]
```diff
@@ -27,68 +21,36 @@ function __init__()
...
 
-Return the array of paths where `matrices`'s groups will be or were downloaded.
+Download the matrix with name `name` in group `group`.
+Return the path where the matrix is stored.
 """
-function group_paths(matrices; format="MM")
-  format ∈ ssmc_formats || error("unknown format $format")
-  joinpath.(ssmc_dir, format, matrices.group)
+function fetch_ssmc(group::AbstractString, name::AbstractString; format="MM")
+  group_and_name = group * "/" * name * "." * format
```
Comment (amontoison) [inline src/SuiteSparseMatrixCollection.jl:33]: Is it working on Windows ?
`\\` is used instead of `/` for this OS.

**Review by amontoison — COMMENTED** [review]
Nice !
Except a minor comment for Windows, everything is fine for me.

File: src/SuiteSparseMatrixCollection.jl (hunk 1 above)
Comment (dpo) [reply src/SuiteSparseMatrixCollection.jl:33]: That's not a path. It's the matrix's full name.

**Comment by dpo** [comment]
So is this better than what we have now?

One the minus side:

* the Artifacts.toml file weighs in at 1.5MB!!!
* we can't download matrices in mat format (not supported by artifacts).

On the plus side:

* it seems some download/untar problems disappeared;
* the code is much shorter.

**Comment by amontoison** [comment]
It's better than what we have now.

In 2021, a file of 1.5MB is not a big deal.
Matrices stored in mat format are the worst because the variable names are not always the same, that's no great loss....

File: utils/gen_artifacts_toml.jl [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+const ssmc_url = "https://sparse.tamu.edu"
+const ssmc_jld2 = joinpath(@__DIR__, "..", "src", "ssmc.jld2")
+db = jldopen(ssmc_jld2, "r")
+matrices = db["df"]
+
+# mat files are not recognized as artifacts
+formats = ("MM", "RB")
+nmatrices = size(matrices, 1) * length(formats)
+const artifacts_toml = joinpath(@__DIR__, "..", "Artifacts.toml")
```
Comment (dpo) [inline utils/gen_artifacts_toml.jl:15]: I just added `@__DIR__` here.

**Review by amontoison — APPROVED** [review]

**Pull request merged**